### PR TITLE
Improve how board and its version is handled by CMake

### PR DESCRIPTION
**Pull request by dragomirecky**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,12 +93,41 @@ set(WUI
...
+  endif()
+endif()
+
+# set MCU to its default if not specified
+if(${MCU} STREQUAL "<default>")
+  set(MCU "STM32F407VG")
+endif()
+
+# parse board version into its components
+string(REGEX MATCH "^([0-9]+)\\.([0-9]+)\\.([0-9]+)" BOARD_VERSION_MATCH ${BOARD_VERSION})
```
Comment (DRracer) [inline CMakeLists.txt:126]: :+1:

File: include/board.h [hunk 2]
```diff
@@ -1,11 +1,50 @@
 #pragma once
+//
```
Comment (DRracer) [inline include/board.h:2]: please prefer doxygen format `///` when writing comments

**Review by DRracer — CHANGES_REQUESTED** [review]
please prefer doxygen format `///` when writing comments, otherwise ok

File: include/board.h (hunk 2 above)
Comment (dragomirecky) [reply include/board.h]: Changed to a doxygen-styled comment

**Review by DRracer — APPROVED** [review]

**Pull request merged**